Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -60,9 +60,15 @@ class TrialOrIntroPriceEligibilityChecker: TrialOrIntroPriceEligibilityCheckerTy
             return
         }
 
+        let completionBlock: ReceiveIntroEligibilityBlock = { result in
+            self.operationDispatcher.dispatchOnMainActor {
+                completion(result)
+            }
+        }
```

Comment:
Extracting and wrapping the completion block from the async call was the only way I found not to have to mark `ReceiveIntroEligibilityBlock` as `@Sendable` up to the public API thus making a breaking change.